gammapy/background/ring.py

```diff
@@ -210,45 +210,35 @@ def run(self, images):
             Result sky maps.
         """
         required = ['counts', 'exposure_on', 'exclusion']
-        counts_map, exposure_on_map, exclusion_map = [images[_] for _ in required]
-
-        result = {}
-        result['exposure_off'] = exposure_on_map.copy(unit='')
-        result['off'] = exposure_on_map.copy(unit='')
-        result['alpha'] = exposure_on_map.copy(unit='')
-        result['background'] = exposure_on_map.copy(unit='')
-
-        for idx in np.ndindex(counts_map.geom.shape):
-            counts = counts_map.get_image_by_idx(idx)
-            exposure_on = exposure_on_map.get_image_by_idx(idx)
-            exclusion = exclusion_map.get_image_by_idx(idx)
-
-            cubes = {}
-
-            kernels = self.kernels(counts)
-
-            cubes['exposure_on'] = self._exposure_on_cube(exposure_on, kernels)
-            cubes['exposure_off'] = self._exposure_off_cube(exposure_on, exclusion, kernels)
+        counts, exposure_on, exclusion = [images[_] for _ in required]
 
-            cubes['off'] = self._off_cube(counts, exclusion, kernels)
-            cubes['alpha_approx'] = self._alpha_approx_cube(cubes)
+        if not counts.geom.is_image:
+            raise NotImplementedError('Adaptive ring background estimation only'
```

Comment:
In other places we always raised ValueError for functions that only work on 2D images. Do it here as well for uniformity?

Reply:
Done